tst/shaka_scheme/system/vm/unit-Compiler.cpp

```diff
@@ -47,3 +48,219 @@ TEST(CompilerUnitTest, quote_test) {
   std::string output = ss.str();
```

Comment:
Did you verify that the input `constant_obj_pair` is actually the proper list `(quote "Hello")` and not the improper list (pair) `(quote . "Hello")` If it happens to be the latter, that would explain the questionable call to `cdr()` used to retrieve the `obj_data` that I commented on above. Please check this if you haven't already. @dylannakahodo

Reply:
@Mortrax You're right. I was constructing it incorrectly, I'll fix this.